dtschema/schemas/system-idle-states.yaml

```diff
@@ -0,0 +1,55 @@
+# SPDX-License-Identifier: BSD-2-Clause
+# Copyright 2025 BayLibre
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/system-idle-states.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: System Idle States
+
+maintainers:
+  - Markus Schneider-Pargmann <msp@baylibre.com>
+
+description:
+  A system idle state node represents a system state that can have certain
+  limitations and power properties.
+
+properties:
+  $nodename:
+    const: system-idle-states
+
+patternProperties:
+  "^system-[a-z0-9-]+$":
+    type: object
+    additionalProperties: false
+    description:
+      The node represents a system state description.
+
+    properties:
+      compatible:
+        const: system-idle-state
+
+      idle-state-name:
+        $ref: /schemas/types.yaml#/definitions/string
+        description:
+          Name of the system state.
```

Comment:
Expanding my previous review. Do the actual values here matter? Can anyone use "pony", "suspend" or "suspend2ram" and be fine? I am asking (again) because I had impression you expect very specific values and if any other value is used, this will break.

That's different than existing meaning of idle-state-name, because before it was just like label, so purely informational.

Reply:
The actual value matters as it's effectively an ABI, it should be the same as described in the datasheet. In my implementation one driver uses that value and checks if it has a specific value (in that case it checks for 'Partial-IO'). I am also describing other states for completeness which are not used by any driver yet. https://lore.kernel.org/lkml/20250421-topic-am62-dt-partialio-v6-15-v1-2-6ced30aafddb@baylibre.com/

I could make this a list of possible values to restrict the possible strings here. But the names are probably mostly SoC family specific so they may have a meaning for one SoC family but would be invalid for another. Would that work for you?